### PR TITLE
Event names & charged_at

### PR DESCRIPTION
**Pull request by mib32**
https://trello.com/c/QSDPlNDn/468-reinstall-app-setup-flow

### CODE REVIEW

File: lib/pr/common/engine.rb [hunk 1]
```diff
@@ -11,6 +11,8 @@ class Engine < ::Rails::Engine
         unless app.root.to_s.match root.to_s
           config.paths["db/migrate"].expanded.each do |expanded_path|
             app.config.paths["db/migrate"] << expanded_path
+
+            ActiveRecord::Migrator.migrations_paths << expanded_path
```
Comment (shkm) [inline lib/pr/common/engine.rb]: Does this do anything other than attempt to fix the error from the other PR? If not, I'd remove it, as I didn't notice a difference.

File: app/controllers/charges_controller.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def create
         redirect_to "#{Settings.client_url}/charge/failed"
       end
     else
-      @user.update(active_charge: true)
+      @user.update(active_charge: true, charged_at: DateTime.now)
```
Comment (shkm) [inline app/controllers/charges_controller.rb]: Can we switch all of the DateTime.now to DateTime.current so we have timezone data?

File: lib/pr/common/shopify_service.rb [hunk 3]
```diff
@@ -29,10 +29,11 @@ def update_shop(plan_name:, uninstalled:)
       def set_uninstalled
         Analytics.track({
                             user_id: @user.id,
-                            event: 'Uninstalled Shopify app',
+                            event: 'App Uninstalled',
                             properties: {
                                 activeCharge: @user.has_active_charge?,
-                                email: @user.email
+                                email: @user.email,
+                                subscription_length: @user.subscription_length
```
Comment (shkm) [inline lib/pr/common/shopify_service.rb]: We don't use subscription_length anywhere else in common. Can you add it to a migration? Ideally with a check in the migration to not do anything if it exists.

File: lib/pr/common/engine.rb (hunk 1 above)
Comment (mib32) [reply lib/pr/common/engine.rb]: @shkm I tested and this helps

File: lib/pr/common/shopify_service.rb (hunk 3 above)
Comment (mib32) [reply lib/pr/common/shopify_service.rb]: @shkm It is a method on a user.rb see above

File: app/controllers/charges_controller.rb (hunk 2 above)
Comment (mib32) [reply app/controllers/charges_controller.rb]: @shkm Why do we need it here?

Comment (shkm) [reply app/controllers/charges_controller.rb]: You know, in our case I don't think it matters much. Ignore!

**Pull request merged**